### ISSUE TITLE
use custom checker callbacks in expectJSONTypes only validates against top level fields

### ISSUE THREAD

**Issue by marcoscarceles** [vlucas/frisby#197]
Hi,

first of all, great work on building this framewok. I found it really useful.

In the docs you mention custom checker callbacks and use them like here:

```javascript
    in_reply_to_screen_name: function(val) { expect(val).toBeTypeOrNull(String); }, // Custom matcher callback
```

I've noticed, however, that when these functions are applied to fields deeper on the JSON structure, instead of the function being invoked it is treated as checking for type Object. Please see the example below:

```javascript
/*
Assuming a response like
{
  "id" : "abc",
  "_links" : {
    "self" : {
      "href" : "/api"
    }
  }
}
*/


var frisby = require('frisby');

var baseURL = "http://localhost:8080"

frisby.create('This fails when it should fail (expected behaviour)')
  .get(baseURL+'/api')
  .expectStatus(200)
  .expectJSONTypes({
    "id" : function(val) { expect(false).toBe(true); },
    "_links" : {
      "self" : Object
    }
  })
  .toss();


frisby.create('This passes when it should fail')
  .get(baseURL+'/api')
  .expectStatus(200)
  .expectJSONTypes({
    "id" : String,
    "_links" : {
      "self" : function(val) { expect(false).toBe(true); }
    }
  })
  .toss();

frisby.create('This fails for the wrong reason, it\'s comparing string and object')
  .get(baseURL+'/api')
  .expectStatus(200)
  .expectJSONTypes({
    "id" : String,
    "_links" : {
      "self" : {
        "href" : function(val) { expect(false).toBe(true); }
      }
    }
  })
  .toss();
```
https://gist.github.com/marcos-carceles/1d622aba90ede3ae4ce7

**Comment by marcos-carceles** [comment]
Hi @vlucas,

I raised this issue some time ago (and submitted PR #199 which fixes it). Could you please give me an update on it?

Cheers!

**Comment by lacyrhoades** [comment]
Confirming the deep callbacks don't work without something like #199 - Thanks!

**Comment by vlucas** [comment]
I just merged in #199 and will publish a new release soon.
